build.sbt

```diff
@@ -57,6 +57,7 @@ lazy val io               = Project("daffodil-io", file("daffodil-io")).configs(
 lazy val runtime1         = Project("daffodil-runtime1", file("daffodil-runtime1")).configs(IntegrationTest)
                               .dependsOn(io, lib % "test->test", udf, macroLib % "compile-internal, test-internal")
                               .settings(commonSettings, usesMacros)
+                              .settings(libraryDependencies ++= Dependencies.exi)
```

Comment:
I wonder if there is value is splitting all of our infoset inputters/outputters into separate subprojects with separate dependencies? That way our runtime doesn't depend on a bunch of different libraries, many of which wouldn't be used.

This means API users would need to add a new dependency depending on which infoset they want to use (e.g. daffodil-runtime1-infoset-xml, daffodil-runtime-infoset-json, daffodil-runtime1-infoset-sax), but it makes runtime and it's dependencies smaller.

We could still have the CLI depend on all or a subset of these for which ever infoset inputters we want to make available.

Reply:
If we do this, please use new distinct packages for each such component, and those packages should NOT span module boundaries. This is a requirement for Java Modules/OGSI packaging that our existing modules and packages are not adhering to, and we need to fix (separate JIRA ticket), but,... no point in making it worse.

Reply:
I'm not sure that I like the idea of moving those 3 infoset inputters/outputter into a separate library.  I think Daffodil should have some subset of infoset handlers built in (the most common, which I would assume to be these 3) and then perhaps the other less common infoset implementations could be set up as separate libraries?  I know it's not exactly the most consistent design, but it would feel strange to me to not include such basic functionality in the main Daffodil library.

Reply:
Agreed. We could maybe take advantage of optional dependencies for this. That way runtime1 would depend on all of infoset subprojects, but a subset of the less common ones would be marked as optional--any API users that want to use an optional one would need to explicitly add them. This would make it trivial to change the "core" set of infoset inputters/outputters by just marking some as required vs optional.

Reply:
Also, I'm not suggesting this be done for this PR. This is a separate issue that requires a bit more thought. This PR just sparked this idea of a potential improvement to me.